lib/ansible/plugins/action/package.py

```diff
@@ -45,14 +46,20 @@ def run(self, tmp=None, task_vars=None):
         if module == 'auto':
             try:
                 if self._task.delegate_to:  # if we delegate, we should use delegated host's facts
-                    module = self._templar.template("{{hostvars['%s']['ansible_facts']['pkg_mgr']}}" % self._task.delegate_to)
+                    module = C.config.get_config_value('PACKAGE_MANAGER_OVERRIDE', variables=combine_vars(self._task.vars, task_vars.get('delegated_vars', {})))
+                    if module is None:
+                        dh = '%s' % self._task.delegate_to
+                        module = self._templar.template(f"{{hostvars['{dh}']['ansible_local']['pkg_mgr']|default(hostvars['{dh}']['ansible_facts']['pkg_mgr']})}}"
                 else:
-                    module = self._templar.template('{{ansible_facts.pkg_mgr}}')
+                    module = C.config.get_config_value('PACKAGE_MANAGER_OVERRIDE', variables=task_vars)
+                    if module is None:
+                        module = self._templar.template('{{ansible_local.pkg_mgr|default(ansible_facts.pkg_mgr)}}')
```

Comment:
Something I forgot about `ansible_local`, is that is namespaces things by filename.  So we might have to do something like a file named `pkg_mgr.fact` with contents of `{"value": "foo"}` and use `ansible_local.pkg_mgr.value`

Reply:
or name it override? can be expanded to more things, like service_mgr